config/monitor_services_test.go

```diff
@@ -453,3 +453,27 @@ func TestServicesMonitorConfig_GoString(t *testing.T) {
 		})
 	}
 }
+
+// TestServicesMonitorConfig_RegexpNil tests the exception that when `Regexp` is
+// unset, it retains nil value after Finalize() and Validate(). Tests that
+// that is it idempotent
```

Comment:
nit: change `Tests that that is it idempotent` to `Tests that Regexp is idempotent`